src/sparql/reports/COUNT-all_diseases_synonym_stats.sparql

```diff
@@ -36,21 +40,32 @@ WHERE {
            owl:annotatedTarget ?synonym ;
            oboInOwl:hasDbXref ?xref .
 
-    BIND(
-      IF(STRSTARTS(STR(?xref), "OMIM:"), "exact_OMIM",
-      IF(STRSTARTS(STR(?xref), "Orphanet:"), "exact_Orphanet",
-      IF(STRSTARTS(STR(?xref), "NCIT:"), "exact_NCIT",
-      IF(STRSTARTS(STR(?xref), "DOID:"), "exact_DO",
-      IF(STRSTARTS(STR(?xref), "ICD10CM:"), "exact_ICD10CM",
-      IF(STRSTARTS(STR(?xref), "icd11.foundation:"), "exact_ICD11", "exact_other")))))) AS ?category)
+    # Map known xref prefixes to categories
+    VALUES (?xref_prefix ?category) {
+      ("OMIM" "exact_OMIM")
+      ("Orphanet" "exact_Orphanet")
+      ("NCIT" "exact_NCIT")
+      ("DOID" "exact_DOID")
+      ("ICD10CM" "exact_ICD10CM")
+      ("MEDGEN" "exact_MEDGEN")
+      ("UMLS" "exact_UMLS")
+      ("NORD" "exact_NORD")
+      ("ClinGen" "exact_ClinGen")
+      ("icd11.foundation" "exact_ICD11")
+      ("OMIA" "exact_OMIA")
+      ("SCTID" "exact_SCTID")
+    }
 
-    FILTER(?category != "exact_other")
+    BIND(STRBEFORE(STR(?xref), ":") AS ?actual_prefix)
+    FILTER(?actual_prefix = ?xref_prefix)
```

Comment:
Never used it like this (FILTER with a constraint on a key of a VALUES statement) but you would know if it does not work!